Speaking mode

reidash opened the pull request:
Adding of Speaking mode

Commit:
cordova plugin add https://github.com/macdonst/SpeechRecognitionPlugin
it requires this plugin for the speaking mode to work
I think need some help with fixing the UI bug because I am still unable to fix that weird bug

Commit:
forgot to push package.json

Commit:
Correcting plugin name in package.json

Commit:
UI bug fix

Commit:
Merge branch 'master' of https://github.com/reidash/Phonetics into Speaking-mode

Commit:
Enabling speaking mode levels

Commit:
Getting rid of console log for testing

Commit:
Cleaning up of the speaking mode

Commit:
Merge branch 'Speaking-mode' of https://github.com/reidash/Phonetics into Speaking-mode

Commit:
Cleaning of the code, removing of console logs and testing comments

cindyzunic commented on Project/src/pages/SpeakingMode/SpeakingMode.ts:
```diff
@@ -0,0 +1,189 @@
+    platform: any;
+    title: string; // Title of the session
+    isCorrect: any;
+    public currUnit: screenUnit; // Current screenUnit
```
Don't think these need to be "public." They're not public in the listening mode file.

cindyzunic commented on Project/src/pages/SpeakingMode/SpeakingMode.ts:
```diff
@@ -0,0 +1,189 @@
+                            this.autoAdvance();
+                            return;
+                        }
+						//means out of all the options, non is correct
```
Can you fix the weird spacing here? And the extra new lines below? Sorry, I'm OCD... haha...

reidash replied:
Okay done!

cindyzunic commented on Project/src/pages/SpeakingMode/SpeakingMode.ts:
```diff
@@ -0,0 +1,189 @@
+        this.currUnit = this.screenUnits[this.currIndex]; // Set current screenUnit
+
+        let randomIndex = Math.floor(Math.random() * this.currUnit.audioPaths.length); // Pick audio clip to use
+        this.plt.ready().then((readySource) => { // Make sure the platform is ready before we try to use native components
```
Have a look at ListeningMode.ts. In ListeningMode, I moved this this.plt.ready() check into the constructor. If we leave it in initUnit, we're doing a platform check every time we init a new screen unit, but we don't need to. You can do the same thing here as we do in ListeningMode: 

1. Do the this.plt.ready().... check in the constructor and set a variable "this.loaded = true" when platform is ready.
2. Setup SpeakingMode.html like ListeningMode.html where you have:

```
<ion-content *ngIf="!loaded">
  <div class="loader">Loading...</div>
</ion-content>
```

and then the second <ion-content> (the real content) would have `*ngIf="loaded"`

This way, we only check platform once, and we don't allow the user to interact with anything until platform is ready. Let me know if you need any help!

Commit:
Made changes to SpeakingMode
- edited the formatting of some parts of the code
- change the position of platform.ready to constructor
- edited html

cindyzunic commented on Project/src/pages/SpeakingMode/SpeakingMode.ts:
```diff
@@ -0,0 +1,184 @@
+    public screenUnits: screenUnit[] = []; // Array of all screen units in the session
+
+    constructor(public navCtrl: NavController, public navParams: NavParams, public plt: Platform, private toastCtrl: ToastController, private zone: NgZone) {
+        this.platform = plt;
```
Why are you setting this.platform = plt here but using this.plt below?

Commit:
Further cleanup, explicitly declaring variables private

Commit:
Whitespace

Commit:
Getting rid of some todo comments
